drivers/spi/spi_nrfx_spi.c

```diff
@@ -432,8 +436,10 @@ static int spi_nrfx_init(const struct device *dev)
 			    nrfx_isr, nrfx_spi_##idx##_irq_handler, 0);	       \
 	}								       \
 	static struct spi_nrfx_data spi_##idx##_data = {		       \
-		SPI_CONTEXT_INIT_LOCK(spi_##idx##_data, ctx),		       \
-		SPI_CONTEXT_INIT_SYNC(spi_##idx##_data, ctx),		       \
+		IF_ENABLED(CONFIG_MULTITHREADING,			       \
+			(SPI_CONTEXT_INIT_LOCK(spi_##idx##_data, ctx),))       \
+		IF_ENABLED(CONFIG_MULTITHREADING,			       \
+			(SPI_CONTEXT_INIT_SYNC(spi_##idx##_data, ctx),))       \
```

Comment:
How about using empty macros `SPI_CONTEXT_INIT_LOCK()` and `SPI_CONTEXT_INIT_SYNC()` when multithreading is disabled? Such changes in drivers wouldn't be necessary then.